index.js

```diff
@@ -6,6 +6,22 @@ import pRetry from 'p-retry';
 // Matches '/<re/po>/tree/<ref>/<dir>'
 const urlParserRegex = /^[/]([^/]+)[/]([^/]+)[/]tree[/]([^/]+)[/](.*)/;
 
+async function isResponseLfs(response) {
+	const length = Number.parseInt(await response.headers.get('content-length'), 10);
+	if (length > 128 && length < 140) {
+		const text = await response.text();
+		return text.startsWith('version https://git-lfs.github.com/spec/v1');
```

Comment:
This function will _use_ the response and then return false. I think you need:

```suggestion
		const contents = await response.clone().text();
		return contents.startsWith('version https://git-lfs.github.com/spec/v1');
```